Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-FIPFSScanEngineVersionState.ps1

```diff
@@ -0,0 +1,147 @@
+
+                $pipeline2ReturnObject = ([PSCustomObject]@{
+                        isE15    = $false
+                        Affected = $false
+                    })
+
+                try {
+                    [int]$fileMajor = $pipeline2FileInfo.VersionInfo.FileMajorPart
+                    [int]$fileMinor = $pipeline2FileInfo.VersionInfo.FileMinorPart
+                    [int]$fileBuild = $pipeline2FileInfo.VersionInfo.FileBuildPart
+                    [int]$filePrivate = $pipeline2FileInfo.VersionInfo.FilePrivatePart
+
+                    if ($fileMajor -eq 15) {
+
+                        switch ($fileMinor) {
+                            0 { $pipeline2ReturnObject.isE15 = $true }
+                            1 { $pipeline2ReturnObject.Affected = (($fileBuild -le 2375) -and ($filePrivate -le 17)) }
+                            2 { $pipeline2ReturnObject.Affected = (($fileBuild -le 986) -and ($filePrivate -le 14)) }
+                            Default { Write-Verbose "Unexpected minor passed to the switch statement" }
+                        }
+                    } else {
+                        Write-Verbose "Exchange server version is not affected by the FIP-FS update issue"
+                    }
+                } catch {
+                    Write-Verbose "Error occured while detecting pipeline2.dll version"
+                    & $CatchActionFunction
+                }
+            }
+            return $pipeline2ReturnObject
+        }
+
+        function GetHighestScanEngineVersionNumber {
+            param (
+                [string]$ComputerName,
+                [scriptblock]$CatchActionFunction
+            )
+
+            Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+
+            try {
+                $scanEngineVersions = Invoke-ScriptBlockHandler -ComputerName $ComputerName `
+                    -ScriptBlock { Get-ChildItem -Path (Join-Path $env:ExchangeInstallPath "FIP-FS\Data\Engines\amd64\Microsoft\Bin") -Attributes Directory } `
+                    -CatchActionFunction $CatchActionFunction
+
+                if ($null -ne $scanEngineVersions) {
+                    [Int64]$highestScanEngineVersion = ($scanEngineVersions.Name | Measure-Object -Maximum).Maximum
+                } else {
+                    Write-Verbose "No FIP-FS scan engine version(s) detected"
+                }
+            } catch {
+                Write-Verbose "Error occured while processing FIP-FS scan engine version(s)"
+                & $CatchActionFunction
+            }
+            return $highestScanEngineVersion
+        }
+    } process {
+        $isAffectedByFIPFSUpdateIssue = $false
+        try {
+            $pipeline2 = TestPipeline2Version -ComputerName $ComputerName `
+                -CatchActionFunction $CatchActionFunction
+
+            if ($pipeline2.Affected -or
+                $pipeline2.isE15) {
```

Comment:
I thought this doesn't affect 2013 (E15) so why are we entering this logic if we are 2013?

Reply:
We've recently added the following to the blog post and so we decided to validate E15 as well:

```
We have an Exchange 2013 server and while there are no crashes, I see the server has the problem engine version starting with "22...". What should we do?
Exchange Server 2013 is not impacted by transport crashes so there will be no buildup of email in transport queues. If your Exchange 2013 server took the antimalware update and it is now on version starting with "22..." you should use the automated or manual steps in this blog post to get your server on an engine version "21..." to continue getting the antimalware updates. Without taking action your server will not get any future antimalware updates.
```